### PR TITLE
Feature: document type validation

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Validate document type in meta data file](https://tools.hmcts.net/jira/browse/BPS-267)

### Change description ###

Re-introducing almost same changes as were in #298 . I believe every1 is in favour of having enum in place. After few takes and thoughts adding validation just to schema didn't feel right. So to "perhaps" assure the "non-breaking" change - including the migration script in the PR to patch up `document_type` historically.

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/resources/1111006_2.metadata.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
...
         "file_name": "1111006.pdf",
         "notes": "",
-        "document_type": "SSC1"
+        "document_type": "Cherished"
```
Comment (nitinprabhu) [inline src/functionalTest/resources/1111006_2.metadata.json]: Unfortunately this is going to change now and we are going to get document type(Other, Cherished and SSCS1) and document subtype with same value initially. Later Exela is going to send only Other and Cherished and everything else goes into subtype. This is awaiting confirmation from Exela.

Comment (doncem) [reply src/functionalTest/resources/1111006_2.metadata.json]: Enum can be extended without a sweat. Let's get final confirmation even though task has already been set

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/model/mapper/EnvelopeResponseMapperTest.java [hunk 2]
```diff
@@ -15,18 +16,21 @@
...
     @Test
-    public void should_map_envelope_to_envelope_response() throws Exception {
+    public void should_map_envelope_to_envelope_response() {
         EnvelopeResponse response = EnvelopeResponseMapper.toEnvelopeResponse(envelope);
 
         assertThat(response)
             .usingComparatorForFields(
                 new ToStringComparator<Classification>(), new String[]{"classification"}
             )
+            .usingComparatorForFields(
```
Comment (lgonczar) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/model/mapper/EnvelopeResponseMapperTest.java]: Is this needed?

Comment (doncem) [reply src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/model/mapper/EnvelopeResponseMapperTest.java]: yes. response is mapped to string value whereas assertion is done against domain model which has enum there and assertion does not know how to convert to string automatically. same treatment as `classification` here

**Pull request merged**